pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1270,6 +1279,10 @@ func (s *SlirpBindMechanism) setCachedInterface(pid string) error {
 	return nil
 }
 
+func (b *SlirpBindMechanism) hasCachedInterface() bool {
+	return true
+}
+
```

Comment:
For this kind of default interface implementations you can try to create a defaultImpl and embed it to the BindMechanism implementations so they have a default implementation of all the methods.

```golang
package main                                                                    
                                                                                
import "fmt"                                                                    
                                                                                
type iface interface {                                                          
    isFoo() bool                                                                
    isBar() bool                                                                
}                                                                               
                                                                                
type defaultImpl struct{}                                                       
                                                                                
func (d *defaultImpl) isFoo() bool {                                            
    return true                                                                 
}                                                                               
                                                                                
func (d *defaultImpl) isBar() bool {                                            
    return false                                                                
}                                                                               
                                                                                
func (d *defaultImpl) isDar() bool {                                            
    return false                                                                
}                                                                               
                                                                                
type final struct {                                                             
    defaultImpl // Embed the defaultImpl to the final impl so it needs to implement just isFoo                                                                 
}                                                                               
                                                                                
func (f *final) isFoo() bool {                                                  
    return false                                                                
}                                                                               
                                                                                
func main() {                                                                   
    f := final{}                                                                
    fmt.Println(f.isFoo())                                                      
    fmt.Println(f.isBar())                                                      
    fmt.Println(f.isDar())                                                      
}                                         
```

Reply:
The current proposal is strictly temporary; once I have the new `VMNetworkingConfigurator` implementations in place I can change the bind mechanism interface (and ultimately rename it). Once I can, I'll separate everything related from phase#1 from the bind mechanism.

Slirp does not do anything on phase#1 - as such, it will not implement any of its methods. IMO, there's no point in writing a default implementation for something that will never be used. I'd rather invest my time in splitting the interfaces, and make sure each bind method (not mechanism) implements what it needs, instead of a one size for all approach.